### DISCUSSION TITLE
op names

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#778] [Feature Requests]
this is probably blasphemy but it bugs me for ages. but maybe its worth to trigger a discussion for the cables future op naming

i find it still not intuitive that op names start with its type, looking for GetString results in this
![image](https://github.com/cables-gl/cables_docs/assets/1138673/a6807e5b-8f4d-4f00-99cb-baa923901f9e)

op start with its type, rather than with the function, takes always a little longer (at least for me) to make sure i take the right one. inside the patch, that rule makes it even harder to quickly distinguish the ops because you end up with a lot of arraysomething op. and it makes the op names unnecessarily longer than they need to be. because we already have color coding for types and in the node browser it also says Ops.Array in small letters. It's a little bit doppeltgemoppelt

an alternative could be 
(op browser)
GetNumber (ops.array)
GetNumber (ops.Json)
GetNumber (ops.vars)

and in the patchfield, you have just shorter op names with color coding

another advantage, apart from shorter op names is, you only need to learn one name for a certain function and apply this name to different types, without having to type it and read it in the patch, again and again.

From my experience with students, they learn ops faster if ops are named the same regarding their function. my brain works the same, i usually think of "getnumber" first and then of the type i need.

this comment comes obviously from other patching tools, so yes...i'm biased. but there are some arguments regarding shorter names without types.

oh, one thing... there are ops without type in its name, its a bit inconsitent right now
![image](https://github.com/cables-gl/cables_docs/assets/1138673/e32d4994-948b-4bd9-a43c-bbb19a137da0)

**Comment by wirmachenbunt** [comment]
digging deeper, there is more
![image](https://github.com/cables-gl/cables_docs/assets/1138673/bb7e137b-ba65-4d19-b586-848f74f99bd2)
randomnumberarray does not have array first, looks like the naming is random :)

**Comment by bejucoo** [comment]
It is true that color coding would be a good option to differentiate ops with the same function. Another option could be to have "just one" op in the search result. For example just one GetNumber and after you click you can select from which kind of type: var, obj, arr, etc. But maybe a bit clunky and extra steps.
